### PR TITLE
VMClone create mutator: Don't mutate when there is no need

### PR DESCRIPTION
**Pull request by orelmisan**
### What this PR does
Before this PR:
Currently, the VMClone create mutator creates a JSON patch to replace the admitted object's spec even when there is no need to replace it (spec.target is fully set).

After this PR:
VMClone create mutator does not add a JSON patch to the AdmissionResponse in case there is no need to mutate the object.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [X] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by vasiliy-ul — APPROVED** [review]
Looks okay from my perspective.

**Review by fossedihelm — APPROVED** [review]

File: pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go [hunk 1]
```diff
@@ -62,16 +63,22 @@ func (mutator *CloneCreateMutator) Mutate(ar *admissionv1.AdmissionReview) *admi
...
-	if err != nil {
+	if err := json.Unmarshal(ar.Request.Object.Raw, &vmCloneOrig); err != nil {
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
+	vmClone := vmCloneOrig.DeepCopy()
+
 	mutateClone(vmClone, mutator.targetSuffix)
 
+	if reflect.DeepEqual(vmClone, vmCloneOrig) {
```
Comment (acardace) [inline pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go:76]: I'm a little bit hesitant to both using the reflect package (added dependency) and using `DeepEqual` as it usually comes with a non-negligible performance impact.

Either way, I don't think you need to compare the whole object, it's sufficient to compare the 2 Specs as the patch only replace the Spec section.

@xpivarc can you take a look as well? It'd be interesting to see if we're getting a speed-up from this or the opposite.

Comment (vasiliy-ul) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go:76]: ... And one thing that I overlooked: if we decide to go with `DeepEqual` it should rather be `equality.Semantic.DeepEqual`. The implementation in `reflect` may not produce expected results.

Comment (xpivarc) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go:76]: In future (hopefully this year), we should have a tools to scale test this. I would wait if there is no immediate benefit

Comment (orelmisan) [reply pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go]: Replaced the call to `reflect.DeepCopy` with a function to detect whether the target was mutated.

**Comment by orelmisan** [comment]
[Change](https://github.com/kubevirt/kubevirt/compare/0397844ae3f0b71eacd6d9f5d755b37217127c3a..c3a9212038bf53a5e908ef3fed5dda777bebdd89): Answered @acardace's [comment](https://github.com/kubevirt/kubevirt/pull/11682#discussion_r1560030560).

**Review by fossedihelm — APPROVED** [review]
Thank you!

**Pull request merged**